Rename SO_{RCV,SND}TIMEO for Linux 5.1

https://github.com/ponylang/ponyc/blob/2bf65c0ed5975e97c7ef28ad242ead294b10ff05/src/libponyrt/lang/socket.c#L4473-L4474
https://github.com/ponylang/ponyc/blob/2bf65c0ed5975e97c7ef28ad242ead294b10ff05/src/libponyrt/lang/socket.c#L4530-L4531

They have been [renamed](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=45bdc66159d49bfc7f75fe02d25bc74f5d2660cf) in Linux 5.1.

This will still fail to compile on Linux 5.1+:
```
In file included from src/libponyrt/lang/socket.c:49:
src/libponyrt/lang/socket.c: In function 'pony_os_sockopt_option':
src/libponyrt/lang/socket.c:4474:22: error: 'SO_RCVTIMEO_OLD' undeclared (first use in this function); did you mean 'SO_RCVTIMEO_NEW'?
 4474 |     case 887: return SO_RCVTIMEO;
      |                      ^~~~~~~~~~~
src/libponyrt/lang/socket.c:4474:22: note: each undeclared identifier is reported only once for each function it appears in
src/libponyrt/lang/socket.c:4531:22: error: 'SO_SNDTIMEO_OLD' undeclared (first use in this function); did you mean 'SO_SNDTIMEO_NEW'?
 4531 |     case 906: return SO_SNDTIMEO;
      |                      ^~~~~~~~~~~
make: *** [Makefile:941: build/release/obj-native/libponyrt/lang/socket.o] Error 1
```
Renaming them with `_OLD` works fine for me.

--- a/packages/net/ossockopt.pony
+++ b/packages/net/ossockopt.pony
@@ -1320,3 +1320,7 @@ primitive OSSockOpt
   fun udp_no_check6_rx():I32 => @pony_os_sockopt_option(I32(1095))
   fun udp_no_check6_tx():I32 => @pony_os_sockopt_option(I32(1096))
   fun udp_vendor():I32 => @pony_os_sockopt_option(I32(1097))
+  fun so_rcvtimeo_old():I32 => @pony_os_sockopt_option(I32(1098))
+  fun so_rcvtimeo_new():I32 => @pony_os_sockopt_option(I32(1099))
+  fun so_sndtimeo_old():I32 => @pony_os_sockopt_option(I32(1100))
+  fun so_sndtimeo_new():I32 => @pony_os_sockopt_option(I32(1101))

--- a/src/libponyrt/lang/socket.c
+++ b/src/libponyrt/lang/socket.c
@@ -5103,6 +5103,18 @@ PONY_API int pony_os_sockopt_option(int option)
 #ifdef UDP_VENDOR
     case 1097: return UDP_VENDOR;
 #endif
+#ifdef SO_RCVTIMEO_OLD
+    case 1098: return SO_RCVTIMEO_OLD;
+#endif
+#ifdef SO_RCVTIMEO_NEW
+    case 1099: return SO_RCVTIMEO_NEW;
+#endif
+#ifdef SO_SNDTIMEO_OLD
+    case 1100: return SO_SNDTIMEO_OLD;
+#endif
+#ifdef SO_SNDTIMEO_NEW
+    case 1101: return SO_SNDTIMEO_NEW;
+#endif
     default: return -1;
   }
 }

